src/components/MovieProfile/index.js

```diff
@@ -0,0 +1,125 @@
+import React, { useState } from "react";
+import { useParams } from "react-router-dom";
+
+import GenrePill from "../GenrePill";
+import Trailer from "../Trailer";
+import Icon from "../shared/Icon";
+import getIcon from "../../util/getIcon";
+import getRuntime from "../../util/getRuntime";
+import convertDate from "../../util/convertDate";
+import {
+    Profile,
+    ProfileContainer,
+    Backdrop,
+    Content,
+    Genres,
+    Stats,
+    Stat,
+    MovieButton,
+    Poster,
+} from "./styles";
+import useDetails from "../../hooks/useDetails";
+
+export default function MovieProfile({ id }) {
+    const type = 'movie';
+    const { data: details, isLoading, isError } = useDetails(id, type);
+    const [isTrailerOpen, setIsTrailerOpen] = useState(false);
+
+    const handleClick = () => setIsTrailerOpen(true);
+
+    if (isLoading) return "Movie Loading...";
+
+    if (isError) return "Movie Error";
+
+    document.title = `MovieZone - ${
+        details.name ? details.name : details.title || details.original_title
+    }`;
+
+    return (
+        <>
+            <Trailer
+                id={id}
+                isOpen={isTrailerOpen}
+                setIsTrailerOpen={setIsTrailerOpen}
+                type={type}
+            />
+
+            <Profile>
+                {details.backdrop_path && (
+                    <Backdrop
+                        src={`https://image.tmdb.org/t/p/original/${details.backdrop_path}`}
+                        alt="backdrop"
+                    />
+                )}
+                <ProfileContainer>
+                    {details.poster_path && (
+                        <Poster
+                            src={`https://image.tmdb.org/t/p/w200/${details.poster_path}`}
+                            alt="poster"
+                        />
+                    )}
+                    <Content>
+                        <h2>
+                            {details.name
+                                ? details.name
+                                : details.original_title}
+                        </h2>
+                        <Genres>
+                            {details.genres &&
+                                details.genres.map((genre) => (
```

Comment:
How bout using optional operators like so:
```javascript
details.genres?.map()
```

instead of
```javascript
details.genres && details.genres.map()
```